src/cc65/declare.c

```diff
@@ -483,21 +555,116 @@ static void ParseEnumDecl (void)
 
         /* Check for an assigned value */
         if (CurTok.Tok == TOK_ASSIGN) {
+
             ExprDesc Expr;
             NextToken ();
             ConstAbsIntExpr (hie1, &Expr);
-            EnumVal = Expr.IVal;
+            EnumVal       = Expr.IVal;
+            MemberType    = Expr.Type;
+            IsSigned      = IsSignSigned (MemberType);
+            IsIncremented = 0;
+
+        } else {
+
+             /* Defaulted with the same signedness as the previous member's */
+            IsSigned = IsSignSigned (MemberType) &&
+                       (unsigned long)EnumVal != GetIntegerTypeMax (MemberType);
+
+            /* Enumerate. Signed integer overflow is UB but unsigned integers
+            ** are guaranteed to wrap around.
+            */
+            EnumVal = (long)((unsigned long)EnumVal + 1UL);
+
+            if (UnqualifiedType (MemberType->C) == T_ULONG && EnumVal == 0) {
+                /* Warn on 'unsigned long' overflow in enumeration */
+                Warning ("Enumerator '%s' overflows the range of '%s'",
+                         Ident,
+                         GetBasicTypeName (type_ulong));
+            }
+
+            IsIncremented = 1;
+        }
+
+        /* Track down the min/max values and evaluate the type of EnumVal
+        ** using GetEnumeratorType in a tricky way.
+        */
+        if (!IsSigned || EnumVal >= 0) {
+            if ((unsigned long)EnumVal > MaxConstant) {
+                MaxConstant = (unsigned long)EnumVal;
+            }
+            NewType = GetEnumeratorType (0, EnumVal, IsSigned);
+        } else {
+            if (EnumVal < MinConstant) {
+                MinConstant = EnumVal;
+            }
+            NewType = GetEnumeratorType (EnumVal, 0, 1);
         }
 
-        /* Add an entry to the symbol table */
-        AddConstSym (Ident, type_int, SC_ENUMERATOR | SC_CONST, EnumVal++);
+        /* GetEnumeratorType above should never fail, but just in case */
+        if (NewType == 0) {
+            Internal ("Unexpected failure with GetEnumeratorType: %lx", EnumVal);
+            NewType = type_ulong;
+        } else if (SizeOf (NewType) < SizeOf (type_int)) {
+            /* Integer constants are not shorter than int */
+            NewType = type_int;
+        }
+
+        /* Warn if the incremented value exceeds the range of the previous
+        ** type.
+        */
+        if (IsIncremented   &&
+            EnumVal >= 0    &&
```

Comment:
Is this type of aligning of operators to a tab stop(?) a style seen in cc65?

Reply:
Yeah. I saw that frequently in the optimizer.... It's aligning to the logic AND/OR to the same tab stop "grids" according to their "levels".